File: leetcode/69. X的平方根/leetcode_69.go
package main

import "fmt"

func mySqrt(x int) int {
	low, high, res := 0, x, -1
	for low <= high {
		mid := low + (high-low)/2
		if mid*mid <= x {
			res = mid
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return res
}

func main() {
	fmt.Println(mySqrt(8))
}

File: leetcode/69. X的平方根/leetcode_69.py
def mySqrt(x: int) -> int:
    # if x in [0, 1]:
    #     return x

    low, high, res = 0, x, -1
    while low <= high:
        mid = (low + high) // 2
        if mid * mid <= x:
            res = mid
            low = mid + 1
        else:
            high = mid - 1
    return res


if __name__ == '__main__':
    print(mySqrt(8))
